interceptor/tracing_interceptor.go

```diff
@@ -0,0 +1,476 @@
+		ToHeader:  true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var finishOpts TracerFinishSpanOptions
+	defer span.Finish(&finishOpts)
+
+	run, err := t.Next.SignalWithStartWorkflow(ctx, in)
+	finishOpts.Error = err
+	return run, err
+}
+
+type tracingActivityInboundInterceptor struct {
+	ActivityInboundInterceptorBase
+	tracer Tracer
+}
+
+func (t *tracingActivityInboundInterceptor) ExecuteActivity(
+	ctx context.Context,
+	in *ExecuteActivityInput,
+) (interface{}, error) {
+	// Start span reading from header
+	info := activity.GetInfo(ctx)
+	span, ctx, err := startSpanFromContext(ctx, t.tracer, &TracerStartSpanOptions{
+		Operation:  "RunActivity",
+		Name:       info.ActivityType.Name,
+		DependedOn: true,
+		Tags: map[string]string{
+			workflowIDTagKey: info.WorkflowExecution.ID,
+			runIDTagKey:      info.WorkflowExecution.RunID,
+		},
+		FromHeader: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var finishOpts TracerFinishSpanOptions
+	defer span.Finish(&finishOpts)
+
+	ret, err := t.Next.ExecuteActivity(ctx, in)
+	finishOpts.Error = err
+	return ret, err
+}
+
+type tracingWorkflowInboundInterceptor struct {
+	WorkflowInboundInterceptorBase
+	tracer Tracer
+}
+
+func (t *tracingWorkflowInboundInterceptor) Init(outbound WorkflowOutboundInterceptor) error {
+	i := &tracingWorkflowOutboundInterceptor{tracer: t.tracer}
+	i.Next = outbound
+	return t.Next.Init(i)
+}
+
+func (t *tracingWorkflowInboundInterceptor) ExecuteWorkflow(
+	ctx workflow.Context,
+	in *ExecuteWorkflowInput,
+) (interface{}, error) {
+	// Start span reading from header
+	info := workflow.GetInfo(ctx)
+	span, ctx, err := startSpanFromWorkflowContext(ctx, t.tracer, &TracerStartSpanOptions{
+		Operation: "RunWorkflow",
+		Name:      info.WorkflowType.Name,
+		Tags: map[string]string{
+			workflowIDTagKey: info.WorkflowExecution.ID,
+			runIDTagKey:      info.WorkflowExecution.RunID,
+		},
+		FromHeader: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var finishOpts TracerFinishSpanOptions
+	defer span.Finish(&finishOpts)
+
+	ret, err := t.Next.ExecuteWorkflow(ctx, in)
+	finishOpts.Error = err
+	return ret, err
+}
+
+type tracingWorkflowOutboundInterceptor struct {
+	WorkflowOutboundInterceptorBase
+	tracer Tracer
+}
+
+func (t *tracingWorkflowOutboundInterceptor) ExecuteActivity(
+	ctx workflow.Context,
+	activityType string,
+	args ...interface{},
+) workflow.Future {
+	// Start span writing to header
+	span, ctx, err := t.startNonReplaySpan(ctx, "StartActivity", activityType, true)
+	if err != nil {
+		return err
+	}
+	defer span.Finish(&TracerFinishSpanOptions{})
+
+	return t.Next.ExecuteActivity(ctx, activityType, args...)
+}
+
+func (t *tracingWorkflowOutboundInterceptor) ExecuteLocalActivity(
+	ctx workflow.Context,
+	activityType string,
+	args ...interface{},
+) workflow.Future {
+	// Start span writing to header
+	span, ctx, err := t.startNonReplaySpan(ctx, "StartActivity", activityType, true)
+	if err != nil {
+		return err
+	}
```

Comment:
There should probably be something that distinguishes a local activity from a normal one. A tag or a different name or something

Reply:
I believe in both Node and Java this is just `StartActivity` with no special tags for local vs non-local (granted Node has no "local" concept yet). Is this something we want to ask all SDKs to differentiate?

Reply:
Marking resolved, but open to general cross-SDK approach to saying whether span represents local activity or not